src/authenticators/direct_authenticator/mod.rs

```diff
@@ -13,18 +13,19 @@ use super::Authenticate;
 use log::error;
 use parsec_interface::requests::request::RequestAuth;
 use parsec_interface::requests::{ResponseStatus, Result};
+use secrecy::ExposeSecret;
 use std::str;
 
 #[derive(Copy, Clone, Debug)]
 pub struct DirectAuthenticator;
 
 impl Authenticate for DirectAuthenticator {
     fn authenticate(&self, auth: &RequestAuth) -> Result<ApplicationName> {
-        if auth.is_empty() {
+        if auth.buffer.expose_secret().is_empty() {
             error!("The direct authenticator does not expect empty authentication values.");
             Err(ResponseStatus::AuthenticationError)
         } else {
-            match str::from_utf8(auth.bytes()) {
+            match str::from_utf8(auth.buffer.expose_secret()) {
                 Ok(str) => Ok(ApplicationName(String::from(str))),
```

Comment:
Since you're doing this change anyway, I'm wondering if it would make sense to put the `String` inside `ApplicationName` into a `Secret` too. The app name sort of is a client-level secret...

I guess the alternative would be to derive the `zeroize(drop)` when the change for that is made at some point.

Thoughts?

Reply:
I am wondering if the application name, similarly to the key name, is a secret. In our TM, it's only the authentication token which is one. I don't know if an attacker could do anything with this information.

Reply:
Which made me realise something - we need to somehow determine which applications use which authenticator. Imagine the following scenario: a Parsec service that has a JWT token authenticator and a direct authenticator (for some reason) - an attacker finds the name of an application and issues requests using direct authentication. Essentially, if there are two or more authenticators, an attacker will always target the easier one to break, if possible.

I think it's right not to consider the app name a secret (even though for direct authentication that *is* the auth token).